base_layer/core/src/transactions/transaction_protocol/single_receiver.rs

```diff
@@ -104,7 +104,8 @@ impl SingleReceiverTransactionProtocol {
                 .construct_proof(spending_key, sender_info.amount.into())?
         };
 
-        let sender_features = sender_info.features.clone();
+        let sender_features =
```

Comment:
Why would the recovery byte not be correct here already?

Reply:
The spending key was not known before this point; this is the receiver building an output with the suggested features from the sender.